resources/sts/4.14/hypershift/sts_extended_hcp_support_permission_policy.json

```diff
@@ -1,115 +1,407 @@
+      "Resource": "*"
+    },
+    {
+      "Sid": "DescribeVpcAttribute",
+      "Effect": "Allow",
+      "Action": [
+        "ec2:DescribeVpcAttribute"
+      ],
+      "Resource": "arn:aws:ec2:*:*:vpc/*",
+      "Condition": {
+        "StringEquals": {
+          "aws:ResourceTag/red-hat-managed": "true"
+        }
+      }
+    },
+    {
+      "Sid": "DescribeVpcEndpointServicePermission",
+      "Effect": "Allow",
+      "Action": [
+        "ec2:DescribeVpcEndpointServicePermissions"
+      ],
+      "Resource": "arn:aws:ec2:*:*:vpc-endpoint-service/*"
+    },
+    {
+      "Sid": "GetS3objectMetadata",
+      "Effect": "Allow",
+      "Action": [
+        "s3:GetObjectAcl",
+        "s3:GetObjectTagging"
+      ],
+      "Resource": "arn:aws:s3:::*",
+      "Condition": {
+        "StringEquals": {
+          "aws:ResourceTag/red-hat-managed": "true"
+        }
+      }
+    },
+    {
+      "Sid": "GetAndCreateTirosQueries",
+      "Effect": "Allow",
+      "Action": [
+        "tiros:CreateQuery",
+        "tiros:GetQueryAnswer",
+        "tiros:GetQueryExplanation"
+      ],
+      "Resource": "*"
+    },
+    {
+      "Sid": "CreateRedHatInstanceTag",
+      "Effect": "Allow",
+      "Action": [
+        "ec2:CreateTags"
+      ],
+      "Resource": "arn:aws:ec2:*:*:instance/*",
+      "Condition": {
+        "StringEquals": {
+          "ec2:CreateAction": "RunInstances",
+          "aws:RequestTag/red-hat-managed": "true"
+        },
+        "ForAllValues:StringEquals": {
+          "aws:TagKeys": "red-hat-managed"
+        }
+      }
+    },
+    {
+      "Sid": "NetworkVerifierInstance",
+      "Effect": "Allow",
+      "Action": [
+        "ec2:RunInstances",
+        "ec2:DescribeInstanceTypes",
+        "ec2:GetConsoleOutput"
+      ],
+      "Resource": "arn:aws:ec2:*:*:instance/*",
+      "Condition": {
+        "StringEquals": {
+          "aws:ResourceTag/red-hat-managed": "true"
+        }
+      }
+    },
+    {
+      "Sid": "DescribeIp",
+      "Effect": "Allow",
+      "Action": [
+        "ec2:DescribeCoipPools",
+        "ec2:DescribeIpv6Pools",
+        "ec2:DescribePublicIpv4Pools",
+        "ec2:GetAssociatedIpv6PoolCidrs"
+      ],
+      "Resource": "*"
+    },
+    {
+      "Sid": "DescribeInternetGateway",
+      "Effect": "Allow",
+      "Action": [
+        "ec2:DescribeEgressOnlyInternetGateways",
+        "ec2:DescribeInternetGateways"
+      ],
+      "Resource": "*"
+    },
+    {
+      "Sid": "DescribePrefixLists",
+      "Effect": "Allow",
+      "Action": [
+        "ec2:DescribePrefixLists",
+        "ec2:DescribeManagedPrefixLists"
+      ],
+      "Resource": "*"
+    },
+    {
+      "Sid": "GetManagedPrefixListsEntries",
+      "Effect": "Allow",
+      "Action": [
+        "ec2:GetManagedPrefixListEntries"
+      ],
+      "Resource": "arn:aws:ec2:*:*:prefix-list/*"
+    },
+    {
+      "Sid": "DescribeImages",
+      "Effect": "Allow",
+      "Action": [
+        "ec2:DescribeImages"
```

Comment:
What images are these used for? Can they be restricted to red-hat-managed: true tag?

Reply:
this can be used to describe the ami the instances use. Since the worker nodes in HCP use public AMI, there's no re-hat-managed: true tag 
`
            "CreationDate": "2023-10-21T05:50:11.000Z",
            "ImageId": "ami-00d973a79002d742b",
            "ImageLocation": "531415883065/rhcos-414.92.202310210434-0-x86_64",
            "ImageType": "machine",
            "Public": true,
            "OwnerId": "531415883065",
            "PlatformDetails": "Linux/UNIX",
            "UsageOperation": "RunInstances",
            "State": "available",
            "BlockDeviceMappings": [
                {
                    "DeviceName": "/dev/xvda",
                    "Ebs": {
                        "DeleteOnTermination": true,
                        "SnapshotId": "snap-01720b82512455a1c",
                        "VolumeSize": 16,
                        "VolumeType": "gp2",
                        "Encrypted": false
                    }
                },
                {
                    "DeviceName": "/dev/xvdb",
                    "VirtualName": "ephemeral0"
                }
            ],
            "Description": "OpenShift 4 414.92.202310210434-0 x86_64",
            "EnaSupport": true,
            "Hypervisor": "xen",
            "Name": "rhcos-414.92.202310210434-0-x86_64",
            "RootDeviceName": "/dev/xvda",
            "RootDeviceType": "ebs",
            "SriovNetSupport": "simple",
            "VirtualizationType": "hvm",
            "DeprecationTime": "2025-10-21T05:50:11.000Z"
        }
    ]
}
`